src/server/pfs/server/master.go

```diff
@@ -34,9 +42,122 @@ func (d *driver) master(ctx context.Context) {
 			gc := chunk.NewGC(d.storage.ChunkStorage())
 			return gc.RunForever(ctx)
 		})
+		eg.Go(func() error {
+			return d.finishCommits(ctx)
+		})
 		return eg.Wait()
 	}, backoff.NewInfiniteBackOff(), func(err error, _ time.Duration) error {
 		log.Errorf("error in pfs master: %v", err)
 		return nil
 	})
 }
+
+func (d *driver) finishCommits(ctx context.Context) error {
+	return d.commits.ReadOnly(ctx).WatchF(func(ev *watch.Event) error {
```

Comment:
Does this spit out everything at the start, and then diffs?  We need to make sure that if we miss an event, everything still gets compacted.

Reply:
Watch sets up the listen, does a get, then syncs the get with the notifications to ensure that events are not lost.